lib/ansible/modules/net_tools/basics/read_mail.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, Caio Ramos <caioramos97@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: read_mail
+
+short_description: Module for reading emails from a IMAP server
+
+version_added: "2.9"
+
+description:
+    - "Get emails from a IMAP server using basic filtering rules an returns an email count and list"
+
+options:
+    host:
+        description:
+            - IMAP server host
+        required: false
+        default: localhost
+    port:
+        description:
+            - IMAP server port (defaults to 993 or 143 when less then 0)
+        required: false
+        default: -1
+    username:
+        description:
+            - Username to get the emails
+        required: true
+    password:
+        description:
+            - Password to get the emails
+        required: true
+    ssl:
+        description:
+            - Whether to use SSL or not
+        required: false
+        default: true
+    mailbox:
+        description:
+            - Select the mailbox to use
+        required: false
+        default: INBOX
+    filter:
+        description:
+            - Default IMAP filters. Checkout the options at RFC 3501 section 6.4.4.
+        required: false
+        default: {'ALL': ''}
+
+author:
+    - Caio Ramos (@caiohsramos)
+    - Gabriely Rangel (@gprangel)
+'''
+
+EXAMPLES = '''
+# simplest use
+- name: Read mails
+  read_mail:
+    host: imap.myhost.com
+    password: mypass
+
+# no ssl and custom username and port
+- name: Read mails
+  read_mail:
+    host: imap.myhost.com
+    port: 42
+    username: myuser
+    password: mypass
+    ssl: no
+
+# using filters
+- name: Read mails with 'subject' and 'to' filters
+  read_mail:
+    host: imap.myhost.com
+    port: 42
+    username: myuser
+    password: mypass
+    filter:
+      subject: mysubject
+      to: example@example.com
+
+- name: Read unseen mails with 'subject' filter
+  read_mail:
+    host: imap.myhost.com
+    username: myuser
+    password: mypass
+    filter:
+      subject: mysubject
+      unseen:
+'''
+
+RETURN = '''
+host:
+    description: The original host param that was passed in
+    type: str
+    returned: always
+username:
+    description: The username used in the request
+    type: str
+    returned: always
+mails:
```

Comment:
if just the mail headers are returned: where do I obtain the mail body itself?

Reply:
I'll add a field with the email's body and think of some use cases for that